src/orquestra/sdk/_ray/_build_workflow.py

```diff
@@ -235,16 +235,19 @@ def _ray_remote(*inner_args, **inner_kwargs):
                 # True for all the steps except data aggregation
                 serialization = user_fn_ref is not None
 
-                user_fn = _get_user_function(user_fn_ref, dry_run, output_metadata)
-
-                wrapped = ArgumentUnwrapper(
-                    user_fn=user_fn,
-                    args_artifact_nodes=args_artifact_nodes,
-                    kwargs_artifact_nodes=kwargs_artifact_nodes,
-                    deserialize=serialization,
-                )
+                try:
+                    user_fn = _get_user_function(user_fn_ref, dry_run, output_metadata)
+
+                    wrapped = ArgumentUnwrapper(
+                        user_fn=user_fn,
+                        args_artifact_nodes=args_artifact_nodes,
+                        kwargs_artifact_nodes=kwargs_artifact_nodes,
+                        deserialize=serialization,
+                    )
 
-                wrapped_return = wrapped(*inner_args, **inner_kwargs)
+                    wrapped_return = wrapped(*inner_args, **inner_kwargs)
```

Comment:
Yes, _get_user_function does the unpickling part, where it might happen that side-effects of the __init__ libraries needed to unpickle are causing an exception. The example of this can be benchq library that starts Julia in __init__. If Julia is not installed, the exception would be thrown at 239 line, not 248.